reconcile/utils/terrascript_aws_client.py

```diff
@@ -1863,8 +1861,16 @@ def populate_tf_resource_s3(self, spec):
         }
         values["policy"] = json.dumps(policy, sort_keys=True)
         values["depends_on"] = self.get_dependencies([user_tf_resource])
-        tf_resource = aws_iam_user_policy(identifier, **values)
-        tf_resources.append(tf_resource)
+        tf_aws_iam_policy = aws_iam_policy(identifier, **values)
+        tf_resources.append(tf_aws_iam_policy)
+
+        tf_user_policy_attachment = aws_iam_user_policy_attachment(
+            identifier,
+            user=identifier,
```

Comment:
These values are identical, so there's not much practical difference between referencing the object or the identifier.

I'm a fan of:

```
user=f"{{{user_tf_resource.name}}}",
```

Reply:
This being said, its not a must change or needs to be fixed for me to approve it.